Source/Core/Duality/Utility/Coroutines/CoroutineManager.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Duality.Utility.Coroutines
+{
+	/// <summary>
+	/// The Coroutine Manager
+	/// </summary>
+	public class CoroutineManager
+	{
+		private readonly Queue<Coroutine> pool = new Queue<Coroutine>(64);
+		private readonly Queue<Coroutine> coroutines = new Queue<Coroutine>(256);
+		private readonly Queue<Coroutine> nextCycle = new Queue<Coroutine>(256);
+		private readonly Queue<Coroutine> scheduled = new Queue<Coroutine>();
+		private readonly Queue<Coroutine> trashcan = new Queue<Coroutine>();
+
+		private readonly Dictionary<Coroutine, Exception> lastFrameErrors = new Dictionary<Coroutine, Exception>();
+
+		private int count;
+
+		/// <summary>
+		/// Returns an IEnumerable of all currently active and scheduled Coroutines
+		/// </summary>
+		public IEnumerable<Coroutine> Coroutines
+		{
+			get { return this.coroutines.Concat(this.scheduled); }
+		}
+
+		/// <summary>
+		/// Returns an IEnumerable of the Exception encountered in the last cycle, with the relative Coroutine
+		/// </summary>
+		public IEnumerable<KeyValuePair<Coroutine, Exception>> LastFrameErrors
+		{
+			get { return this.lastFrameErrors; }
+		}
+
+		/// <summary>
+		/// Prepares a new Coroutine and places it in the scheduled queue, to be started in the next cycle
+		/// </summary>
+		/// <param name="enumerator">The Coroutine's execution body</param>
+		/// <param name="name">The name of the Coroutine</param>
+		/// <returns>The prepared Coroutine</returns>
+		internal Coroutine StartNew(IEnumerator<WaitUntil> enumerator, string name)
+		{
+			Coroutine coroutine;
+			if (this.pool.Count > 0)
+				coroutine = this.pool.Dequeue();
+			else
+				coroutine = new Coroutine();
+
+			coroutine.Setup(enumerator, name);
+
+			this.scheduled.Enqueue(coroutine);
+			return coroutine;
+		}
+
+		/// <summary>
+		/// Returns true if the Coroutine is currently in execution or scheduled for execution in the next cycle
+		/// </summary>
+		/// <param name="c">The Coroutine</param>
+		/// <returns>True if found</returns>
+		public bool IsCoroutineListed(Coroutine c)
```

Comment:
This method is not used, do we really need it? If we dont then we don't need the Coroutines property either.

Reply:
Again, just a convenience method for the users to see if a Coroutine is still listed or not... think there's no need for this?

Reply:
I cant think of a usecase for this. We can remove it for the first release and always add it back later on if we need it.